pkg/k8s/resource/resource.go

```diff
@@ -561,7 +561,10 @@ func (r *resource[T]) resourceName() string {
 	// For example, with resource[*cilium_api_v2.CiliumNode] new(T) returns **cilium_api_v2.CiliumNode
 	// and *new(T) is nil. So we create a new pointer using reflect.New()
 	o := *new(T)
-	sourceObj := reflect.New(reflect.TypeOf(o).Elem()).Interface().(T)
+	sourceObj, ok := reflect.New(reflect.TypeOf(o).Elem()).Interface().(T)
+	if !ok {
+		return ""
```

Comment:
I'm a bit concerned about this, since prior this would just panic but now if this has an unexpected type this will quietly return an empty string.

Reply:
I got rid of the `if` statement. 
```golang
sourceObj, _ := reflect.New(reflect.TypeOf(o).Elem()).Interface().(T)
```
In cases like this it is okay to allow people to do the strong empty-ok assertion. This one-liner is self-evidently fine.